### PR TITLE
[Neo Plugins UT] Fix namespace for adding plugin ut tests

### LINKED ISSUE #3287: [Neo Plugins] Fix the Neo Plugins Namespace
**Summary or problem description**

The namespace in the neo plugin is very confusing and conflicts with the class name, so we need to sort out the namespace.

For Instance, 

it is Neo.Plugins; for some plugin, 
ApplicationLogs.Store; for some plugin,
Neo.Consensus for some plugin.

The issue that makes me have to make this change is the namespace of RpcServer is `Neo.Plugins;`, while it has a class of `RpcServer`, which is exactly the name of the plugin. If i want to add UT test for it, it conflics with the UT namespace as `Neo.Plugins.RpcServer.Tests;`, the namespace of the UT makes it impossible to get access to the RpcServer class.

**Do you have any solution you want to propose?**
Now all change to Neo.Plugins.XXXX;

**Where in the software does this update applies to?**
- Plugins

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

The namespace in the neo plugin is very confusing and conflicts with the class name, so we need to sort out the namespace.

For Instance, 

it is Neo.Plugins; for some plugin, 
ApplicationLogs.Store; for some plugin,
Neo.Consensus for some plugin.

Now all change to Neo.Plugins.XXXX;

The issue that makes me have to make this change is the namespace of RpcServer is `Neo.Plugins;`, while it has a class of `RpcServer`, which is exactly the name of the plugin. If i want to add UT test for it, it conflics with the UT namespace as `Neo.Plugins.RpcServer.Tests;`, the namespace of the UT makes it impossible to get access to the RpcServer class.

Fixes # Closes #3287 

## Type of change

- [ ] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
To demonstrate the change, i have added the UT moq that i will be using to test the rpcserver. More tests will be added in other prs.

- [x] TestCheckAuth_ValidCredentials_ReturnsTrue

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shargon** [comment]
@Jim8y any obsolete call that we can remove it? now that the public method will change it's a good time to clean it

**Comment by Jim8y** [comment]
do as you wish.

File: tests/Neo.Plugins.RpcServer.Tests/UT_RpcServer.Blockchain.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using Neo.Ledger;
+using Neo.Persistence;
+using System;
+using System.Text;
+
+namespace Neo.Plugins.RpcServer.Tests
+{
+    public partial class UT_RpcServer
```
Comment (shargon) [inline tests/Neo.Plugins.RpcServer.Tests/UT_RpcServer.Blockchain.cs:22]: Remove it?

Comment (Jim8y) [reply tests/Neo.Plugins.RpcServer.Tests/UT_RpcServer.Blockchain.cs:22]: please dont, its a part of my ut plan, showing how i will work on them. this pr is just the first pr of many others. please keep it if it wont cause problem.

**Review by shargon — APPROVED** [review]

**Review by cschuchardt88 — APPROVED** [review]

File: src/Neo/Neo.csproj [hunk 2]
```diff
@@ -27,6 +27,7 @@
   <ItemGroup>
     <InternalsVisibleTo Include="Neo.SmartContract.Testing" />
     <InternalsVisibleTo Include="Neo.SmartContract.TestEngine" />
+    <InternalsVisibleTo Include="Neo.Plugins.RpcServer.Tests" />
```
Comment (vncoelho) [inline src/Neo/Neo.csproj:30]: @Jim8y 

Why just RpcServer here?

**Review by vncoelho — COMMENTED** [review]
The changes looks good. You could had wait more for merging. There are changes on the organization of files and classes, in particular, consensus.

File: src/Neo/Neo.csproj (hunk 2 above)
Comment (Jim8y) [reply src/Neo/Neo.csproj:30]: adding ut for these plugin will be a very huge task, now i can only focus on rpcserver. whoever want to work on other plugin can add by urself.

Comment (vncoelho) [reply src/Neo/Neo.csproj:30]: I mean, why not include `[Neo.Plugins.OracleService.Tests](https://github.com/neo-project/neo/tree/master/tests/Neo.Plugins.OracleService.Tests)` and [Neo.Plugins.Storage.Tests](https://github.com/neo-project/neo/tree/master/tests/Neo.Plugins.Storage.Tests)

What was this need?

**Pull request merged**